contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,312 @@
+    /// It just checks the asset is WETH and emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    /// @param _asset Address of asset to deposit. Has to be WETH.
+    /// @param _amount Amount of assets that were transferred to the strategy by the vault.
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_asset == WETH, "Unsupported asset");
+        depositedWethAccountedFor += _amount;
+        _deposit(_asset, _amount);
+    }
+
+    /// @dev Deposit WETH to this strategy so it can later be staked into a validator.
+    /// @param _asset Address of WETH
+    /// @param _amount Amount of WETH to deposit
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+        /*
+         * We could do a check here that would revert when "_amount % 32 ether != 0". With the idea of
+         * not allowing deposits that will result in WETH sitting on the strategy after all the possible batches
+         * of 32ETH have been staked.
+         * But someone could mess with our strategy by sending some WETH to it. And we might want to deposit just
+         * enough WETH to add it up to 32 so it can be staked. For that reason the check is left out.
+         *
+         * WETH sitting on the strategy won't interfere with the accounting since accounting only operates on ETH.
+         */
+        emit Deposit(_asset, address(0), _amount);
+    }
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 newWeth = wethBalance - depositedWethAccountedFor;
+
+        if (newWeth > 0) {
+            depositedWethAccountedFor = wethBalance;
+
+            _deposit(WETH, newWeth);
+        }
+    }
+
+    /// @notice Withdraw WETH from this contract. Used only if some WETH for is lingering on the contract. That
+    /// can happen when:
+    ///   - the deposit was not a multiple of 32 WETH
+    ///   - someone sent WETH directly to this contract
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    /// @param _recipient Address to receive withdrawn assets
+    /// @param _asset WETH to withdraw
+    /// @param _amount Amount of WETH to withdraw
+    function withdraw(
+        address _recipient,
+        address _asset,
```

Comment:
We never validate this asset. This function should have a `require(_asset == WETH, "Unsupported asset");`.

Reply:
added